src/logics/Logic.cc

```diff
@@ -162,43 +162,59 @@ bool Logic::hasQuotableChars(std::string const & name) const
     return false;
 }
 
+// If the symbol corresponding to the one specified in the arguments would be ambiguous, i.e., either not known by
+// the user or there is a homonymous symbol with the same return type that is not a constant fixed by the language syntax,
+// specify the return sort.
+std::string Logic::disambiguateName(std::string const & protectedName, SRef sortRef, bool isNullary) const {
+    assert(not protectedName.empty());
+    if (not isNullary) {
+        return protectedName;
+    }
+
+    auto isQuoted = [](std::string const &s) { return s.size() > 3 and *s.begin() == '|' and *(s.end()-1) == '|'; };
+    auto name = isQuoted(protectedName) ?
+            std::string_view(protectedName.data()+1, protectedName.size()-2) : std::string_view(protectedName);
```

Comment:
I am a little confused why the name was protected before with `| |` and now we are removing the protection.
Isn't there some workflow when this would not happen?

Reply:
I'll think about it...  Now the workflow is that 
1. protect the name
2. disambiguate the name

and it feels like it'd be better to
1. disambiguate the name
2. protect the name

However, syntactically the protection needs to be inside the disambiguation construct, so it makes more sense to have the protected name already available when disambiguating.  Otherwise we could add quotations inside the concrete string representing the disambiguation, which seems awkward.

More flexibility would be available if disambiguation and protection would be inside a single function, but in the current design protection needs to be called without disambiguation when producing formal parameters for `declare-fun`s, e.g., when printing models.